rest/src/main/scala/io/udash/rest/DefaultRestClient.scala

```diff
@@ -0,0 +1,70 @@
+package io.udash
+package rest
+
+import com.avsystem.commons._
+import com.avsystem.commons.annotation.explicitGenerics
+import com.avsystem.commons.meta.Mapping
+import com.softwaremill.sttp.Uri.QueryFragment.KeyValue
+import com.softwaremill.sttp.Uri.QueryFragmentEncoding
+import com.softwaremill.sttp._
+import io.udash.rest.DefaultSttpBackend.backend
+import io.udash.rest.raw._
+
+object DefaultRestClient {
+  @explicitGenerics def apply[RestApi: RawRest.AsRealRpc : RestMetadata](baseUri: String): RestApi =
+    RawRest.fromHandleRequest[RestApi](asHandleRequest(baseUri))
+
+  /**
+    * Creates a [[io.udash.rest.raw.RawRest.HandleRequest HandleRequest]] function which sends REST requests to
+    * a specified base URI using default HTTP client implementation (sttp).
+    */
+  def asHandleRequest(baseUri: String): RawRest.HandleRequest =
+    asHandleRequest(uri"$baseUri")
+
+  private def toSttpRequest(baseUri: Uri, request: RestRequest): Request[String, Nothing] = {
+    val uri = baseUri |>
+      (u => u.copy(path = u.path ++
+        request.parameters.path.map(_.value))) |>
+      (u => u.copy(queryFragments = u.queryFragments ++
+        request.parameters.query.iterator.map {
+          case (k, QueryValue(v)) => KeyValue(k, v, QueryFragmentEncoding.All, QueryFragmentEncoding.All)
+        }.toList
+      ))
+
+    val contentTypeHeader = request.body.mimeTypeOpt.map {
+      mimeType => (HeaderNames.ContentType, s"$mimeType;charset=utf-8")
+    }
+    val paramHeaders = request.parameters.headers.iterator.map {
+      case (n, HeaderValue(v)) => (n, v)
+    }.toList
+
+    sttp.copy[Id, String, Nothing](
+      method = Method(request.method.name),
+      uri = uri,
+      headers = contentTypeHeader.toList ++ paramHeaders,
```

Comment:
If order does not matter that much, you can swap these and avoid `.toList`.

Reply:
That will be much slower potentially. `contentTypeHeader.toList` has at most one element so the current version will be O(1).